parachain/pallets/control/src/lib.rs

```diff
@@ -45,13 +47,14 @@ pub mod pallet {
 		type WeightInfo: WeightInfo;
 		type MaxUpgradeDataSize: Get<u32>;
 		type CreateAgentOrigin: EnsureOrigin<Self::RuntimeOrigin, Success = MultiLocation>;
+		type DescribeAgentLocation: DescribeLocation;
 	}
 
 	#[pallet::event]
 	#[pallet::generate_deposit(pub(super) fn deposit_event)]
 	pub enum Event<T: Config> {
 		Upgrade { impl_address: H160, impl_code_hash: H256, params_hash: Option<H256> },
-		CreateAgent { agent_id: H256 },
+		CreateAgent { agent_location: VersionedMultiLocation, agent_id: H256 },
```

Comment:
`VersionedMultiLocation` is only really meant to be used for inputs to extrinsics.

```suggestion
		CreateAgent { agent_location: MultiLocation, agent_id: H256 },
```